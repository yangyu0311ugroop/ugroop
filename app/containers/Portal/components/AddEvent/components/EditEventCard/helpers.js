import dotProp from 'dot-prop-immutable';
import get from 'lodash/get';
import { EVENT_VIEW_HELPERS } from 'smartComponents/Node/types/Event/helpers';

export const convertReservation = data => {
  if (!EVENT_VIEW_HELPERS.isTransportation(data)) {
    const eventBooking = get(data, `detail.detail.common.eventBooking`);

    if (!eventBooking) return data;

    return dotProp.merge(data, 'detail', {
      bookingNumber: eventBooking.bookingNumber,
      bookedBy: eventBooking.bookedBy,
      supplier: eventBooking.supplierName,
      supplierPhoneNumber: eventBooking.supplierNumber,
      bookingPersonCount: eventBooking.personCount,
    });
  }

  const detail = get(data, `detail`);

  return dotProp.merge(data, 'detail.detail.common.eventBooking', {
    bookingNumber: detail.bookingNumber,
    bookedBy: detail.bookedBy,
    supplierName: detail.supplier,
    supplierNumber: detail.supplierPhoneNumber,
    personCount: detail.bookingPersonCount,
  });
};

export const EDIT_HELPERS = {
  convertReservation,
};
