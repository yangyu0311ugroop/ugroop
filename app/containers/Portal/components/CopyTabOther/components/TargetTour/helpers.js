const createModel = data => {
  const {
    firstName,
    lastName,
    email,
    dob,
    dobMode,
    namePersonId,
    personType,
    comment,
    note,
    primary,
    phone,
    phoneId,
    ageType,
    honorificTitle,
    orgId,
  } = data;
  return {
    firstName,
    lastName,
    email,
    dob,
    dobMode,
    namePersonId,
    personType,
    comment,
    note,
    primary,
    phone,
    phoneId,
    ageType,
    honorificTitle,
    orgId,
  };
};

const getParticipant = (nodes, { firstName, lastName, email }) =>
  nodes
    .map(node => node.customData)
    .filter(
      person =>
        (person.firstName === firstName && person.lastName === lastName) ||
        (person.email === email && !!person.email),
    );

export const MODEL_HELPERS = {
  createModel,
  getParticipant,
};
