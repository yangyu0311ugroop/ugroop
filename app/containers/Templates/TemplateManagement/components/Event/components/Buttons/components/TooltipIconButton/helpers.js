import { EVENT_CONSTANTS } from 'utils/constants/events';

const showSubIconFn = (type, subtype, props = {}) => {
  const { showSubIcon } = props;

  if (!showSubIcon) return false;

  const exception = [EVENT_CONSTANTS.TRANSPORTATIONS.FLIGHT.type];

  return exception.indexOf(subtype) === -1;
};

export const EVENT_ICON_HELPERS = {
  showSubIconFn,
};
