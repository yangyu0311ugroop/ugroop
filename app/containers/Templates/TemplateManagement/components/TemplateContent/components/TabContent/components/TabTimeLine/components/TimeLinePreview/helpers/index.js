const FirstElement = 0;

function convertToArrayLists(data) {
  const lists = [];
  let index = 1;
  if (data && data.length > 0) {
    // has day 0
    const firstNode = data[FirstElement];
    lists.push({ id: firstNode.id, title: `Day ${index}` });
    let hasNextChildren = firstNode.nextNodes;
    while (hasNextChildren.length > 0) {
      index += 1;
      lists.push({
        id: hasNextChildren[FirstElement].id,
        title: `Day ${index}`,
      });
      hasNextChildren = hasNextChildren[FirstElement].nextNodes;
    }
  }
  return lists;
}

function traverseToLastNode(data) {
  let node = data[FirstElement];
  while (node && node.nextNodes.length > 0) {
    node = node.nextNodes[0];
  }
  return node;
}

export default { convertToArrayLists, traverseToLastNode };
