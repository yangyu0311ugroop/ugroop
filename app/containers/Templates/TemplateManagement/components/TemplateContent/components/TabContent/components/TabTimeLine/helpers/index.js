import _ from 'lodash';

const updateState = (state, filter, keys, values) => {
  // TODO: optimise the code to use deep clone.
  const cloneState = _.cloneDeep(state);
  const index = _.findIndex(cloneState, filter);
  let i = 0;
  if (index !== -1) {
    keys.forEach(key => {
      cloneState[index][key] = values[i];
      i += 1;
    });
  }
  return { cloneState, index };
};

export { updateState };
