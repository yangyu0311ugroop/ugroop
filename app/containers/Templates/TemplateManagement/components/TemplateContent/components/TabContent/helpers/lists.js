// TODO: Why is this file basically a duplicate of datastore/templateManagementStore/helper/lists.js????

import { List } from 'immutable';

const isOpposite = (lastOperation, newOperation) =>
  (lastOperation.action === 'MOVE_AFTER' &&
    newOperation.action === 'MOVE_BEFORE') ||
  (lastOperation.action === 'MOVE_BEFORE' &&
    newOperation.action === 'MOVE_AFTER');

const reverseAction = action =>
  action === 'MOVE_AFTER' ? 'MOVE_BEFORE' : 'MOVE_AFTER';

const isAcrossMove = ({ parentId, toBeMovedParentId }) =>
  toBeMovedParentId && parentId !== toBeMovedParentId;

const isEmpty = list => !List.isList(list) || list.size === 0;

const hasBeenMoved = (list, id) =>
  list.findIndex(({ toBeMovedId }) => toBeMovedId === id) !== -1;

const hasBeenMovedAcross = (list, id) =>
  list.findIndex(
    ({ toBeMovedId, parentId, toBeMovedParentId }) =>
      toBeMovedId === id && toBeMovedParentId && parentId !== toBeMovedParentId,
  ) !== -1;

export const canCancel = (lastOperation, newOperation) => {
  if (isAcrossMove(lastOperation) !== isAcrossMove(newOperation)) return false;

  // move 1 to after 2, then move 2 to after 1, nothing changed so we can cancel
  if (
    lastOperation.id === newOperation.toBeMovedId &&
    lastOperation.toBeMovedId === newOperation.id &&
    lastOperation.action === newOperation.action
  ) {
    return true;
  }

  // move 1 to after 2, then move 1 to before 2
  return (
    lastOperation.id === newOperation.id &&
    lastOperation.toBeMovedId === newOperation.toBeMovedId &&
    isOpposite(lastOperation, newOperation)
  );
};

// toBeMovedId and action must match
export const canOverride = (lastOperation, newOperation) => {
  if (!lastOperation || !lastOperation.id) return false;

  return (
    lastOperation.action === newOperation.action &&
    lastOperation.toBeMovedId === newOperation.toBeMovedId
  );
};

// check if we can reverse the actions
// i.e instead of move 1 to after 2, we move 2 to before 1.
// Result is same, but sometimes we can save 1 operation
const canReverse = (lastOperation, newOperation) =>
  isOpposite(lastOperation, newOperation) &&
  lastOperation.toBeMovedId === newOperation.id &&
  lastOperation.parentId === newOperation.parentId;

const reverse = (lastOperation, newOperation) => ({
  ...newOperation,
  id: newOperation.toBeMovedId,
  toBeMovedId: lastOperation.toBeMovedId,
  action: lastOperation.action,
});

// if operation is not moved across, toBeMoved is not dirty, node at position is dirty, it's better swap
const betterSwap = (list, operation) =>
  !isAcrossMove(operation) &&
  !hasBeenMoved(list, operation.toBeMovedId) &&
  hasBeenMoved(list, operation.id);

const swap = ({ toBeMovedId, id, action, ...operation }) => ({
  ...operation,
  id: toBeMovedId,
  toBeMovedId: id,
  action: reverseAction(action),
});

const push = params => list => {
  // if list is undefined or not a List
  if (!List.isList(list)) return List([params]);

  // if list is empty, simply push new action
  if (list.size === 0) return list.push(params);

  const nextOperation = params;

  const last = list.last();
  // 2 actions cancel each other, simply pop the last action out the pending queue
  if (canCancel(last, nextOperation)) {
    return list.pop();
  }

  // can reverse the action and node
  // i.e instead of move 1 to after 2, we move 2 to before 1.
  // Result is same, but sometimes we can save 1 operation
  if (canReverse(last, nextOperation)) {
    return list.push(reverse(last, nextOperation));
  }

  // otherwise push new action to end of list
  return list.push(nextOperation);
};

const defaultList = () => List();

const notBelongToSameParent = (operation = {}) => (node = {}) =>
  node.parentId !== operation.parentId;
// clear operations belong to same parent
const clear = operation => list =>
  (list || defaultList()).filter(notBelongToSameParent(operation));

const isFirstLevel = (node = {}) => !node.parentId;
// clear second level sections
const clearSections = list => (list || defaultList()).filter(isFirstLevel);

// discard all operations
const clearAll = () => list => (list || defaultList()).clear();

const toArray = list => {
  // if list is undefined or not a List
  if (!List.isList(list)) return [];

  return list.toArray();
};

// 2 adjacent actions can be merged when
// i.e. move 1 to 2 and then move 1 to 3 > move 1 to 3 directly
const optimiseList = list => {
  if (isEmpty(list)) return list;

  let currentList = list;
  let optimisedList = new List();

  while (currentList.size) {
    const lastOperation = optimisedList.last();
    const next = currentList.first();
    let nextOperation = next;

    // if last operation is dirty and this operation is not dirty, it's better swap
    if (betterSwap(optimisedList, next)) {
      nextOperation = swap(next);
    }

    if (canOverride(lastOperation, nextOperation)) {
      optimisedList = optimisedList.pop();
    }

    currentList = currentList.shift();
    optimisedList = push(nextOperation)(optimisedList);
  }

  return optimisedList;
};

const optimise = list => {
  const groups = list.groupBy(({ parentId }) =>
    parentId ? 'root' : 'children',
  );

  return groups.map(optimiseList).reduce((acc, values) => {
    let newAcc = acc;
    values.forEach(value => {
      newAcc = newAcc.push(value);
    });
    return newAcc;
  }, List());
};

export default {
  has: hasBeenMoved,
  hasBeenMoved,
  hasBeenMovedAcross,
  canReverse,
  reverseAction,
  betterSwap,
  swap,
  notBelongToSameParent,
  isFirstLevel,
  push,
  clear,
  clearSections,
  clearAll,
  toArray,
  isEmpty,
  optimise,
  optimiseList,
  default: defaultList,
};
