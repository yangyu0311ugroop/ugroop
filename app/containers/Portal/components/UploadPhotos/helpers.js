import { DEFAULT, UPLOADED } from 'appConstants';
import dotProp from 'dot-prop-immutable';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import { ACTIVITY } from 'utils/modelConstants';

const gridSize = count =>
  LOGIC_HELPERS.switchCase(count, {
    1: { xs: 12 },
    2: { xs: 12, sm: 6 },
    [DEFAULT]: { xs: 12, sm: 6, md: 4, lg: 3 },
  });

const maxWidth = count => LOGIC_HELPERS.ifElse(count <= 2, 'sm', 'md');

const rotateLeftFn = rotate => {
  if (!rotate || typeof rotate !== 'number') return -90;
  if (rotate === -270) return 0;

  return rotate - 90;
};

const rotateRightFn = rotate => {
  if (!rotate || typeof rotate !== 'number') return 90;
  if (rotate === 270) return 0;

  return rotate + 90;
};

const normaliseSubmitData = ({ id, form, uploadedFiles, droppedFiles }) => (
  accu,
  fileId,
) => {
  const file = dotProp.get(droppedFiles, `${fileId}`);

  if (!file) return accu;

  const { status, rotate, errorMessage } = file;
  const photo = dotProp.get(uploadedFiles, `${fileId}.url`);
  const description = dotProp.get(form, `${fileId}.description`);

  if (status !== UPLOADED || errorMessage) {
    return accu;
  }

  return accu.concat({
    type: ACTIVITY,
    parentNodeId: id,
    customData: {
      description,
      photo,
      metaInfo: {
        rotate,
      },
    },
  });
};

export const UPLOAD_PHOTO_UTILS = {
  gridSize,
  maxWidth,
  rotateLeftFn,
  rotateRightFn,
  normaliseSubmitData,
};
