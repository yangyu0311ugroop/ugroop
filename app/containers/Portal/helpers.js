import {
  ADD_ROUTE,
  PHOTO_PREVIEW,
  MANAGE_TABS,
  PROMPT,
  UPLOAD_PHOTOS,
  USER_DATA_STORE,
  ADD_RISK,
  VIEW_RISK,
  ADD_HAZARD,
  CREATE_CHAT_CHANNEL,
  EDIT_CHAT_DESCRIPTION,
  ADD_PEOPLE_IN_CHANNEL,
  DATES_SELECT,
  POPPER_DIALOG,
  HIDE_CHAT_DESCRIPTION,
  LEAVE_CHAT_DESCRIPTION,
  DELETE_CHAT_DESCRIPTION,
  CLONE_TEMPLATE,
  VIEW_EVENT,
  ADD_ROOM,
  VIEW_ROOM,
  ADD_ROOM_PAX,
  CREATE_TOUR_EMAIL,
  ADD_EVENT,
  PRINT_PDF_FORM,
  ADD_TOUR,
  COPY_PARTICIPANT,
  PARTICIPANT_LIST,
  ADD_TAB,
  COPY_TAB_OTHER,
  SHOW_SYSTEM_UPDATE,
  HELP_DIALOG,
  ADD_EDIT_INSURANCE,
} from 'appConstants';
import { DATA_HELPERS } from 'datastore/utils';
import React from 'react';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import { STRING_HELPERS } from 'utils/stringAdditions';

const openPortal = type => (
  portalProps,
  { resaga },
  portalId,
  now = Date.now(),
) => {
  const id = portalId || now;
  resaga.setValue({
    portalIds: DATA_HELPERS.arrayAdd(id),
    portalData: DATA_HELPERS.objectAdd({
      [id]: { type, ...portalProps },
    }),
  });

  return id;
};

const closePortal = (id, { resaga }) => {
  resaga.setValue({
    portalIds: DATA_HELPERS.arrayRemove(id),
    portalData: DATA_HELPERS.objectRemove(id),
  });
};

// convenience functions, optional, can call openPortal(TYPE)(...props) instead
const openAddRoute = openPortal(ADD_ROUTE);
const openAddRisk = openPortal(ADD_RISK);
const openAddHazard = openPortal(ADD_HAZARD);
const openAddRoom = openPortal(ADD_ROOM);
const openAddRoomPax = openPortal(ADD_ROOM_PAX);
const openCreateChatChannel = openPortal(CREATE_CHAT_CHANNEL);
const openEditChatChannel = openPortal(EDIT_CHAT_DESCRIPTION);
const openAddPeopleInChannel = openPortal(ADD_PEOPLE_IN_CHANNEL);
const openArchiveChannelDialog = openPortal(HIDE_CHAT_DESCRIPTION);
const openLeaveChannelDialog = openPortal(LEAVE_CHAT_DESCRIPTION);
const openDeleteChannelDialog = openPortal(DELETE_CHAT_DESCRIPTION);
const openViewEvent = (portalProps, props, portalId, now) =>
  openPortal(VIEW_EVENT)(
    { flightBookingId: null, id: null, selectedId: null, ...portalProps },
    props,
    portalId,
    now,
  );
const openAddTab = openPortal(ADD_TAB);
const openAddEvent = openPortal(ADD_EVENT);
const openAddTour = openPortal(ADD_TOUR);
const openDatesSelect = openPortal(DATES_SELECT);
const openPopperDialog = openPortal(POPPER_DIALOG);
const openViewRisk = openPortal(VIEW_RISK);
const openViewRoom = openPortal(VIEW_ROOM);
const openPhotoPreview = openPortal(PHOTO_PREVIEW);
const openUploadPhotos = openPortal(UPLOAD_PHOTOS);
const openManageTabs = openPortal(MANAGE_TABS);
const prompt = openPortal(PROMPT);
const openCloneMarketTemplate = openPortal(CLONE_TEMPLATE);
const openCreateTourEmail = openPortal(CREATE_TOUR_EMAIL);
const DANGER = 'delete';
const openPrintPdfForm = openPortal(PRINT_PDF_FORM);
const openCopyParticipant = openPortal(COPY_PARTICIPANT);
const openParticipantList = openPortal(PARTICIPANT_LIST);
const openCopyTabOther = openPortal(COPY_TAB_OTHER);
const openHelpDialog = openPortal(HELP_DIALOG);
const showSystemUpdate = openPortal(SHOW_SYSTEM_UPDATE);
const showAddEditInsurance = openPortal(ADD_EDIT_INSURANCE);

const confirmDeletePhoto = ({ count, onConfirm }, props) =>
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      closeOnConfirm: false,
      onConfirm,
      simplifyDialog: true,
      headlineText: (
        <>
          Are you sure you want to delete{' '}
          {LOGIC_HELPERS.ifElse(
            count > 1,
            `the selected ${STRING_HELPERS.pluralise('image', count)}`,
            'this image',
          )}
          ?
        </>
      ),
      dialogTitle: `Delete ${STRING_HELPERS.pluralise('Image', count)}`,
      confirmButton: LOGIC_HELPERS.ifElse(
        count > 1,
        `Delete ${count} Selected ${STRING_HELPERS.pluralise('Image', count)}`,
        'Delete',
      ),
      cancelButton: 'Go Back',
    },
    props,
  );

const confirmCancelUploadPhotos = ({ onConfirm }, props) => {
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      onConfirm,
      simplifyDialog: true,
      headlineText: "If you discard now, you'll lose your progress.",
      dialogTitle: 'Discard Changes?',
      confirmButton: 'Discard',
      cancelButton: 'Go Back',
    },
    props,
  );
};

const promptEventAdded = ({ name, type = 'Event' }, props) => {
  PORTAL_HELPERS.prompt(
    {
      template: 'confirm',
      confirmation: true,
      simplifyDialog: true,
      headlineText: `${LOGIC_HELPERS.ifElse(
        name,
        `"${name}"`,
        type,
      )} has been added successfully.`,
      dialogTitle: 'Success',
      confirmButton: 'OK',
    },
    props,
  );
};

const confirmCancelAddRoute = ({ onConfirm }, props) => {
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      onConfirm,
      simplifyDialog: true,
      headlineText: "If you discard now, you'll lose your progress.",
      dialogTitle: 'Success',
      confirmButton: 'OK',
    },
    props,
  );
};

const confirmDeleteRoute = ({ onConfirm }, props) =>
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      closeOnConfirm: false,
      onConfirm,
      simplifyDialog: true,
      headlineText: 'Are you sure you want to delete this route?',
      dialogTitle: `Delete Route`,
      confirmButton: 'Delete',
      cancelButton: 'Go Back',
    },
    props,
  );

const confirmDeleteBooking = ({ onConfirm }, props) =>
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      closeOnConfirm: false,
      onConfirm,
      simplifyDialog: true,
      headlineText: 'Are you sure you want to delete this flight itinerary?',
      dialogTitle: `Delete flight itinerary`,
      confirmButton: 'Delete',
      cancelButton: 'Go Back',
    },
    props,
  );

const confirmDelete = (
  {
    onConfirm,
    message,
    title,
    closeOnConfirm,
    confirmButton = 'Delete',
    cancelButton = 'Go Back',
  },
  props,
) =>
  PORTAL_HELPERS.prompt(
    {
      simplifyDialog: true,
      template: DANGER,
      closeOnConfirm,
      onConfirm,
      headlineText: message,
      dialogTitle: title,
      confirmButton,
      cancelButton,
    },
    props,
  );

const confirmCustom = (
  {
    onConfirm,
    message,
    title,
    closeOnConfirm,
    confirmButton = 'Continue',
    cancelButton = 'Go Back',
    template = 'confirm',
  },
  props,
) =>
  PORTAL_HELPERS.prompt(
    {
      simplifyDialog: true,
      template,
      closeOnConfirm,
      onConfirm,
      headlineText: message,
      dialogTitle: title,
      confirmButton,
      cancelButton,
    },
    props,
  );

const confirmCancelTransfer = ({ onConfirm }, props) =>
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      closeOnConfirm: false,
      onConfirm,
      simplifyDialog: true,
      headlineText: 'Are you sure you want to cancel pending transfer?',
      dialogTitle: `Cancel Transfer of Ownership`,
      confirmButton: 'Continue',
      cancelButton: 'Go Back',
    },
    props,
  );
export const PORTAL_IDS_SELECTOR = [USER_DATA_STORE, 'portal', 'portalIds'];
export const PORTAL_DATA_SELECTOR = [USER_DATA_STORE, 'portal', 'portalData'];
export const PORTAL_DATA_ID_SELECTOR = ({ portalId }) => [
  USER_DATA_STORE,
  'portal',
  'portalData',
  portalId,
];
const setValue = {
  portalIds: PORTAL_IDS_SELECTOR,
  portalData: PORTAL_DATA_SELECTOR,
};

const close = ({ onClose, portalId }) => {
  LOGIC_HELPERS.ifFunction(onClose, [portalId]);
};

export const PORTAL_HELPERS = {
  openPhotoPreview,
  openAddRoute,
  openAddRisk,
  openAddHazard,
  openAddRoom,
  openAddRoomPax,
  openViewEvent,
  openAddTab,
  openAddEvent,
  openAddTour,
  openDatesSelect,
  openViewRisk,
  openViewRoom,
  openUploadPhotos,
  openManageTabs,
  openCreateChatChannel,
  openEditChatChannel,
  openAddPeopleInChannel,
  openArchiveChannelDialog,
  openLeaveChannelDialog,
  openDeleteChannelDialog,
  openCloneMarketTemplate,
  openCreateTourEmail,
  openPortal,
  closePortal,
  close,
  prompt,
  confirmDeletePhoto,
  confirmDeleteRoute,
  confirmDelete,
  confirmCancelUploadPhotos,
  promptEventAdded,
  confirmCancelAddRoute,
  setValue,
  confirmCancelTransfer,
  openPopperDialog,
  openPrintPdfForm,
  openCopyParticipant,
  confirmCustom,
  openParticipantList,
  openCopyTabOther,
  confirmDeleteBooking,
  openHelpDialog,
  showSystemUpdate,
  showAddEditInsurance,
};
