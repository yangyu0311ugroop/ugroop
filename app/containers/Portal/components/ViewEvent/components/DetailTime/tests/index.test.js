import { shallow } from 'enzyme';
import React from 'react';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import MOMENT_HELPERS from 'utils/helpers/moment';
import { NODE_HELPERS } from 'utils/helpers/nodes';
import { TEST_HELPERS } from 'utils/helpers/testHelpers';
import { DetailTime } from '../index';

describe('<DetailTime />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    classes: {},
    resaga,
  };

  beforeEach(() => {
    jest.clearAllMocks();

    rendered = shallow(<DetailTime {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(DetailTime).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('formatTime()', () => {
    it('should return null', () => {
      expect(instance.formatTime(null)).toBe(null);
    });

    it('should formatTime', () => {
      LOGIC_HELPERS.ifElse = jest.fn();

      instance.formatTime(MOMENT_HELPERS.testInstance);

      TEST_HELPERS.expectCalled(LOGIC_HELPERS.ifElse);
    });
  });

  describe('renderEndTime()', () => {
    it('should return null', () => {
      NODE_HELPERS.withTime = jest.fn(() => null);

      expect(instance.renderEndTime()).toBe(null);
    });

    it('should renderEndTime', () => {
      NODE_HELPERS.withTime = jest.fn(() => true);
      MOMENT_HELPERS.appendZone = jest.fn(() => 'MOMENT_HELPERS.appendZone');

      TEST_HELPERS.expectMatchSnapshot(instance.renderEndTime, [
        MOMENT_HELPERS.testInstance,
      ]);
    });
  });

  describe('renderTime()', () => {
    it('should return null', () => {
      expect(instance.renderTime(null)).toBe(null);
    });

    it('should renderTime', () => {
      instance.renderEndTime = jest.fn(() => 'renderEndTime');
      instance.formatTime = jest.fn(() => 'formatTime');

      TEST_HELPERS.expectDefined(instance.renderTime, [
        MOMENT_HELPERS.testInstance,
      ]);
    });
  });

  describe('renderDayCount()', () => {
    it('should return null', () => {
      MOMENT_HELPERS.dayCount = jest.fn(() => 0);

      expect(instance.renderDayCount()).toBe(false);
    });

    it('should renderDayCount', () => {
      MOMENT_HELPERS.dayCount = jest.fn(() => 2);

      TEST_HELPERS.expectMatchSnapshot(instance.renderDayCount);
    });
  });

  describe('render()', () => {
    it('should render', () => {
      instance.renderTime = jest.fn(() => 'renderTime');
      instance.renderDayCount = jest.fn(() => 'renderDayCount');

      TEST_HELPERS.expectMatchSnapshot(instance.render);
    });

    it('should render', () => {
      instance.renderTime = jest.fn(() => 'renderTime');
      instance.renderDayCount = jest.fn(() => 'renderDayCount');
      rendered.setProps({ hideEndDate: true });

      TEST_HELPERS.expectMatchSnapshot(instance.render);
    });
  });
});
