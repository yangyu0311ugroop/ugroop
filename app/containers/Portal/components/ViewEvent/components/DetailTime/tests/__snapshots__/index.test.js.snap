// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailTime /> render() should render 1`] = `
<div>
  <WithStyles(GridContainer)
    alignItems="center"
    wrap="nowrap"
  >
    <WithStyles(GridItem)
      xs={true}
    >
      <WithStyles(GridContainer)
        direction="column"
      >
        <WithStyles(GridItem)>
          <WithStyles(JTextClass)
            gray={true}
            sm={true}
          >
            Starting
          </WithStyles(JTextClass)>
        </WithStyles(GridItem)>
        renderTime
      </WithStyles(GridContainer)>
    </WithStyles(GridItem)>
    <WithStyles(GridItem)
      xs={true}
    >
      <WithStyles(GridContainer)
        direction="column"
      >
        <WithStyles(GridItem)>
          <WithStyles(JTextClass)
            gray={true}
            sm={true}
          >
            Ending
          </WithStyles(JTextClass)>
        </WithStyles(GridItem)>
        renderTime
      </WithStyles(GridContainer)>
    </WithStyles(GridItem)>
    renderDayCount
  </WithStyles(GridContainer)>
</div>
`;

exports[`<DetailTime /> render() should render 2`] = `
<div>
  <WithStyles(GridContainer)
    alignItems="center"
    wrap="nowrap"
  >
    <WithStyles(GridItem)
      xs={true}
    >
      <WithStyles(GridContainer)
        direction="column"
      >
        <WithStyles(GridItem)>
          <WithStyles(JTextClass)
            gray={true}
            sm={true}
          >
            Starting
          </WithStyles(JTextClass)>
        </WithStyles(GridItem)>
        renderTime
      </WithStyles(GridContainer)>
    </WithStyles(GridItem)>
    <WithStyles(GridItem)
      xs={true}
    />
    renderDayCount
  </WithStyles(GridContainer)>
</div>
`;

exports[`<DetailTime /> renderDayCount() should renderDayCount 1`] = `
<div>
  <WithStyles(GridItem)>
    <WithStyles(GridContainer)
      alignItems="center"
      direction="column"
      spacing={0}
    >
      <WithStyles(GridItem)>
        <WithStyles(JTextClass)
          black={true}
          xxl={true}
        >
          3
        </WithStyles(JTextClass)>
      </WithStyles(GridItem)>
      <WithStyles(GridItem)>
        <WithStyles(JTextClass)
          gray={true}
          sm={true}
          uppercase={true}
        >
          Day
        </WithStyles(JTextClass)>
      </WithStyles(GridItem)>
    </WithStyles(GridContainer)>
  </WithStyles(GridItem)>
</div>
`;

exports[`<DetailTime /> renderEndTime() should renderEndTime 1`] = `
<div>
  <WithStyles(GridItem)>
    <WithStyles(JTextClass)
      danger={true}
      xl={true}
    />
    <WithStyles(JTextClass)
      gray={true}
      sm={true}
    >
      MOMENT_HELPERS.appendZone
    </WithStyles(JTextClass)>
  </WithStyles(GridItem)>
</div>
`;
